twisted/web/test/test_http2.py

```diff
@@ -1403,6 +1403,32 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_resetAfterBody(self):
+        """
+        A client that immediately resets after sending the body causes Twisted
+        to send no response.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        requestBytes = f.preamble()
+        requestBytes += buildRequestBytes(self.getRequestHeaders, [], f)
+        requestBytes += hyperframe.frame.RstStreamFrame(
```

Comment:
I assume that hyperframe.frame.RstStreamFrame is building a reset message, associate it with a stream and pack it as a frame :) 

I was expecting to have the following DSL  `message = h2.makeResetMessage(stream=some_id)`  and then I can do `a.dataReceived(hyperframe.packForWire(message))`